fastlane_core/lib/fastlane_core.rb

```diff
@@ -51,6 +56,21 @@
 module FastlaneCore
   ROOT = Pathname.new(File.expand_path('../..', __FILE__))
 
+  # Session is used to report usage metrics.
+  # If you opt out, we will not send anything.
+  # You can confirm this by observing how we use the environment variable: FASTLANE_OPT_OUT_USAGE
+  # Specifically, in AnalyticsSession.finalize_session
+  # Learn more at https://github.com/fastlane/fastlane#metrics
+  @session = nil
+  def self.session
+    @session ||= AnalyticsSession.new
```

Comment:
If you don't want to allow direct access from the outside to `@session`, the notation is to use `@_session ||= ...`